src/UCTSearch.cpp

```diff
@@ -535,12 +551,16 @@ int UCTSearch::think(int color, passflag_t passflag) {
     // create a sorted list off legal moves (make sure we
     // play something legal and decent even in time trouble)
     float root_eval;
-    if (!m_root->has_children()) {
+    const auto had_children = m_root->has_children();
+    if (m_root->expandable()) {
```

Comment:
Do we care that if a node is expanded multiple times that it may be done so based on a different rotation each time? This would only happen if the cache had pushed the original result out.
I guess there is also a slim possibility that a node's result is pushed out of the cache but its children's results aren't. Hmm, would that have any interesting effects?

Reply:
>This would only happen if the cache is full and had pushed the original result out.

It should be noted that this code only does something if we have a very large tree, so this may be very likely to happen.

Reply:
https://github.com/gcp/leela-zero/commit/bae6d40fbefcef970c57939d0766ebdd5baa4f70#diff-28216771f77a09f4e0782ad5404fdd28R152
The interesting effect is that there is no guarantee that all children will be recovered (even for the root node!) and you may not be reserving enough for the vector when a different symmetry is chosen at re-expansion, because the policy is different. We need to store the symmetry (requires 3 bits ...) to be used when calling get_scored_moves (now get_output) in create_children at re-expansion. Otherwise, we need to move elements m_children to corresponding spots of the nodelist, delete elements with policy <new_min_psa and without visits from the nodelist, then use the nodelist as the new m_children, which is a bit too complicated.